src/System.Net.Security/src/System/Net/Security/SslState.cs

```diff
@@ -1373,15 +1373,22 @@ private void HandleQueuedCallback(ref object queuedStateRequest)
                 return;
             }
             queuedStateRequest = null;
-                        
+
             switch (obj)
             {
                 case LazyAsyncResult lazy:
                     lazy.InvokeCallback();
                     break;
                 case TaskCompletionSource<int> taskCompletionSource when taskCompletionSource.Task.AsyncState != null:
                     Memory<byte> array = (Memory<byte>)taskCompletionSource.Task.AsyncState;
-                    taskCompletionSource.SetResult(CheckOldKeyDecryptedData(array));
+                    try
+                    {
+                        taskCompletionSource.SetResult(CheckOldKeyDecryptedData(array));
```

Comment:
I'm a little confused here. Where is the exception coming from? CheckOldKeyDecryptedData? If so, why isn't the try/catch around that specifically? It seems weird to wrap the SetResult call itself in a try/catch.

Reply:
Agreed. The revised code should look something like this:

```c#
try
{
    TYPE_RETURNED_FROM_CheckOldKeyDecryptedData result = CheckOldKeyDecryptedData(array);
    taskCompletionSource.SetResult(result);
}
catch (Exception e)
{
    taskCompletionSource.SetException(e);
}
```

Reply:
I don't have a strong opinion here: I think what Tomas has is fine, but I'd also be fine changing it.  If we change it, though, it would be to make the only thing in the try block be the CheckOldKeyDecryptedData call, so the above example code does not help.  If we really want to move the SetResult call (which won't throw) out of the try, it would instead look more like:
```C#
int oldKeyResult;
try
{
    oldKeyResult = CheckOldKeyDecryptedData(array);
}
catch (Exception exc)
{
    taskCompletionSource.SetException(exc);
}
taskCompletionSource.SetResult(oldKeyResult);

Reply:
I updated the code. All option seems similar enough to me and it seems like rest is style/personal preference.